routers/api/v2/users.go

```diff
@@ -386,6 +389,50 @@ func (r *apiV2Router) GetPasswordResetEmail(ctx *gin.Context) {
 	ctx.Status(http.StatusOK)
 }
 
+// PUT: /api/v2/users/:id/role
+// x-www-form-urlencoded
+// Request:  role string
+// Headers:  Authorization -> token
+func (r *apiV2Router) SetRole(ctx *gin.Context) {
+	var err error
+
+	role := ctx.PostForm("role")
+	if len(role) == 0 {
+		r.logger.Debug("role not provided")
+		models.SendAPIError(ctx, http.StatusBadRequest, "user role must be provided")
+		return
+	}
+
+	// TODO: Replace auth level conversion once V1 is removed
+	var authLevel v1.AuthLevel
+	err = authLevel.UnmarshalJSON([]byte(strconv.Quote(role)))
+	if err != nil {
+		r.logger.Debug("invalid role", zap.Error(err))
+		models.SendAPIError(ctx, http.StatusBadRequest, "role does not exist")
+		return
+	}
+
+	err = r.userService.UpdateUserWithID(ctx, ctx.Param("id"), services.UserUpdateParams{
+		entities.UserAuthLevel: authLevel,
```

Comment:
Is there a reason why you don't want to create a dedicated field for the role?

Reply:
Nope, not particularly, I'm going to add the role